### ISSUE TITLE
System.DateTime.DayOfYear is wrong for midnight dates

### ISSUE THREAD

**Issue by stopiccot** [bridgedotnet/Bridge#3450]
Pretty self explaining. Midnight dates get rounded to previous day

### Steps To Reproduce

https://deck.net/b6a0879451f92f73c6ac3a33420cc92a

```csharp
public class Program
{
    public static void Main()
    {
        var d = new System.DateTime(2018, 3, 6, 1, 1, 1);
        
        Console.WriteLine(d.DayOfYear);
        Console.WriteLine(d.Date.DayOfYear);
    }
}
```

### Expected Result

```js
65
65
```

### Actual Result

```js
65
64
```

**Comment by geoffreymcgill** [comment]
Hi @stopiccot – Thanks for the report. This should be an easy fix for us to implement.

**Comment by geoffreymcgill** [comment]
Hi @stopiccot – A fix is in place and tested. The updated `DayOfYear` functionality will be included in the upcoming Bridge 17.0 release.

If you're in need of a patch, you can replace the `getDayOfYear` and `getDate` functions in your copy of *bridge.js*, or *date.js* if you're building from source, with the following block:

```js
$clearTime: function (d, isUTC) {
    var dt = new Date(d.getTime());

    if (isUTC === true) {
        dt.setUTCHours(0);
        dt.setUTCMinutes(0);
        dt.setUTCMinutes(0);
        dt.setUTCMilliseconds(0);
    } else {
        dt.setHours(0);
        dt.setMinutes(0);
        dt.setSeconds(0);
        dt.setMilliseconds(0);
    }

    return dt;
},

getDayOfYear: function (d) {
    var dt = System.DateTime.getDate(d),
        ny = new Date(dt);

    if (d.kind === 1) {
        ny.setUTCMonth(0);
        ny.setUTCDate(1);
    } else {
        ny.setMonth(0);
        ny.setDate(1);
    }

    return Math.ceil((dt - ny) / 864e5) + 1;
},

getDate: function (d) {
    d.kind = (d.kind !== undefined) ? d.kind : 0

    var d1 = System.DateTime.$clearTime(d, d.kind === 1);

    d1.kind = d.kind;

    return d1;
},
```

Hope this helps.
